include/ipr/impl

```diff
@@ -1134,6 +1139,17 @@ namespace ipr {
          Category_code operation() const final;
       };
 
+      struct Coerce : Expr<ipr::Coerce> {
+         const ipr::Type& target_type;
+         const ipr::Expr& expression;
+         const ipr::Expr* impl;
+
+         Coerce(const ipr::Type&, const ipr::Expr&);
+         const ipr::Type& target() const final;
+         const ipr::Expr& expr() const final;
+         Optional<ipr::Expr> implementation() const final;
```

Comment:
Coerce represented programmatically define conversions from constructors or conversion functions. This will point to definition. I have updated Coerce to a classic node and revised the comments.